tests/cpp/test_multidevice_overlap.cpp

```diff
@@ -195,6 +195,11 @@ class CollectiveBasedOverlapTest : public OverlapTest {
     tc_locally_reduced_ = at::empty(tc_locally_reduced_sizes, gpu_options_);
   }
 
+  void TearDown() override {
+    tc_locally_reduced_.reset();
```

Comment:
This is a leftover from another draft PR. It's not strictly required but in general things allocated in SetUp should be cleaned in TearDown. If it were allocated in the constructor, we could skip this because the destructor automatically destructs members.